openlibrary/core/wikidata.py

```diff
@@ -0,0 +1,123 @@
+"""
+The purpose of this file is to:
+1. Interact with the Wikidata API
+2. Store the results
+3. Make the results easy to access from other files
+"""
+import requests
+import web
+import dataclasses
+from dataclasses import dataclass
+from openlibrary.core.helpers import days_since
+
+from datetime import datetime
+import json
+from openlibrary.core import db
+
+
+@dataclass
+class WikiDataAPIResponse:
+    type: str
+    labels: dict
+    descriptions: dict
+    aliases: dict
+    statements: dict
+    sitelinks: dict
+    id: str
+
+    @classmethod
+    def from_dict(cls, data: dict):
+        # use [''] instead of get so this fails if fields are missing
+        return cls(
+            type=data['type'],
+            labels=data['labels'],
+            descriptions=data['descriptions'],
+            aliases=data['aliases'],
+            statements=data['statements'],
+            sitelinks=data['sitelinks'],
+            id=data['id'],
+        )
+
+
+@dataclass
+class WikiDataEntity:
+    id: str
+    data: WikiDataAPIResponse
+    updated: datetime
```

Comment:
Let's see if we can combine these two classes ; they only differ by one field, the `updated` field. We might want to rename this to `fetch_time` because then it makes sense in both contexts.

Right now the two very similarly named function are a little dissonant in terms of how close conceptually they are to each other.

Reply:
2 would be preferable :+1: . `updated` works for me!

Reply:
done, and it looks great :)